include/MeshManager.hpp

```diff
@@ -27,21 +27,30 @@ namespace blitzdg {
       index_type NumElements;
       std::string CsvDelimiters;
       real_vec_smart_ptr Vert;
-      index_vec_smart_ptr EToV;
+      index_vec_smart_ptr EToV, BCType;
       index_vec_smart_ptr ElementPartitionMap;
       index_vec_smart_ptr VertexPartitionMap;
 
 
       // Helper method to convert vector of strings to vector of ints
       // representing Gmsh element data.
       std::vector<index_type> parseElem(const std::vector<std::string>& input);
-      
+
+      // Helper method to construct BCType table after element EToV
+      // and edge tables have been parsed.
+      void buildBCTable(std::vector<std::vector<index_type>>& edges);
```

Comment:
It looks like `edges` is not modified by this function, so you should pass it by const reference.

Reply:
Good point, will fix.

Reply:
Actually, at the point that edges is constructed, it is not known _a priori_ the number of edges there are. See `MeshManager::readMesh(const string& gmshInputFile)`, any suggestion on how to deal with this using blitz matrices without just converting to one after building the stack of vector<vector>'s?